kymatio/scattering1d/core/timefrequency_scattering.py

```diff
@@ -64,23 +65,54 @@ def timefrequency_scattering(
     # Apply low-pass filtering over frequency (optional) and unpad
     if average:
         # zero-pad along frequency, map to Fourier domain
-        total_height = 2 ** sc_freq.J_pad
+        total_height = 2 ** sc_freq.J_pad[-1]  # TODO
         S_1_tm_T_hat = _pad_transpose_fft(S_1_list, total_height, B, B.rfft)
 
+        if oversampling_fr == 'auto':
+            # subsample as we would in min-padded case
+            total_subsample_fr_max = sc_freq.J - max(sc_freq.j0s)
+        else:
+            # subsample regularly (relative to current padding)
+            total_subsample_fr_max = sc_freq.J
+
+        if oversampling_fr == 'auto':
+            reference_subsample_equiv_due_to_pad = max(sc_freq.j0s)
+            if out_type == 'array':
+                subsample_equiv_due_to_pad_min = 0
+            elif out_type == 'list':
+                subsample_equiv_due_to_pad_min = (
+                    reference_subsample_equiv_due_to_pad)
+            reference_total_subsample_so_far = (subsample_equiv_due_to_pad_min +
+                                                0)
+        else:
+            reference_total_subsample_so_far = 0
```

Comment:
It's not consistent with `oversampling` (see 4 in my latest comment post)

Reply:
We should rename this. The `+ 0` is there for clarity that a particular value is zero rather than absent, and this variable is shown in `_joint_lowpass` (though I could just explicitly declare it and set to zero; this is the duplication I'm still sorting)